factory/glideFactoryEntry.py

```diff
@@ -1300,6 +1300,19 @@ def unit_work_v3(
                 % (scitoken_file, str(submit_credentials.identity_credentials))
             )
 
+    if ("grid_proxy" in auth_method) or ("scitoken" in auth_method):
+        # Check if project id is required
```

Comment:
Should this be removed? To add the project_id if in auth methods no matter which other auth method is there?

Reply:
Yes, I think it makes sense. If anything we should check the `grid_type` since here https://github.com/glideinWMS/glideinwms/blob/366c9481b81392f5cfb1168257d64cc7516dd928/factory/glideFactoryLib.py#L2043 we add the `project_id` to the rsl and the environment only if the grid type is NOT `ec2`, `gce` and `batch`. But I think it does not really matter since at reconfig time we only use the project id for `condor`: https://github.com/glideinWMS/glideinwms/blob/25e66daac4d0a2a789f1c03617c6f5758dd2c924/creation/lib/cgWCreate.py#L315

So, I would say let's always add it here since it does not hurt. This way we are sure that if `"project_id" in auth_method` then `ProjectId in submit_credential` and avoid the issue of the ticket.

I will make the change, let me know if there is anything else.